### PR TITLE
Refactor printers

### PR DESCRIPTION
**Pull request by vbgl**
This removes the dependency between Printer and PrintLinear modules.

### CODE REVIEW

File: compiler/src/printLinear.ml [hunk 1]
```diff
@@ -11,9 +12,6 @@ module F = Format
 module Pr = Printer
```
Comment (eponier) [inline compiler/src/printLinear.ml:12]: This should go away, I guess?

File: compiler/src/printCommon.ml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  | Ovlsl (ve,ws) -> asprintf "<<%s" (string_of_velem Signed   ws ve)
+
+(* -------------------------------------------------------------------- *)
+
+let pp_var tbl fmt x =
+  let y = Conv.var_of_cvar tbl x in
+  fprintf fmt "%s" y.v_name
+
+let pp_var_i tbl fmt x =
+  pp_var tbl fmt x.E.v_var
```
Comment (eponier) [inline compiler/src/printCommon.ml:92]: This is used only in printLinear.ml, is it worth to put it here?

**Review by eponier — COMMENTED** [review]
2 comments

File: compiler/src/printLinear.ml (hunk 1 above)
Comment (vbgl) [reply compiler/src/printLinear.ml]: Done. Thanks.

File: compiler/src/printCommon.ml (hunk 2 above)
Comment (vbgl) [reply compiler/src/printCommon.ml]: This is in anticipation of a future change. Should I postpone this?

**Review by eponier — APPROVED** [review]

File: compiler/src/printCommon.ml (hunk 2 above)
Comment (eponier) [reply compiler/src/printCommon.ml]: No, that's ok.

**Pull request merged**